### PR TITLE
Add more validations to newly added RUS tables, update metadata

### PR DESCRIPTION
**Pull request by e-belfer**
# Overview

## What did you change?
Add a small number of additional validations to tables merged in #5091, and update to remove WIP/warning label. Needs to merge after #5125.

## Documentation

Make sure to update relevant aspects of the documentation:

- [ ] Update the [release notes](https://catalystcoop-pudl.readthedocs.io/en/nightly/release_notes.html): reference the PR and related issues.
- [ ] Update relevant Data Source jinja templates (see `docs/data_sources/templates`).
- [x] Update relevant table or source description metadata (see `src/metadata`).
- [x] Review and update any other aspects of the documentation that might be affected by this PR.

# Testing

How did you make sure this worked? How can a reviewer verify this?

## To-do list

- [x] If updating analyses or data processing functions: make sure to update row count expectations in `dbt` tests.
- [ ] Run `pixi run pre-commit-run` to run linters and static code analysis checks.
- [ ] Run `pixi run pytest-ci` locally to ensure that the merge queue will accept your PR.
- [x] Review the PR yourself and call out any questions or issues you have.
- [ ] For PRs that change the PUDL outputs significantly, run the full ETL locally and then [run the data validations](https://catalystcoop-pudl.readthedocs.io/en/nightly/dev/data_validation.html) using dbt. If you can't run the ETL locally then run the `build-deploy-pudl` GitHub Action manually and ensure that it succeeds.

### CODE REVIEW

File: dbt/models/rus12/out_rus12__monthly_demand_and_energy_at_power_sources/schema.yml [hunk 1]
```diff
@@ -8,6 +8,10 @@ sources:
               arguments:
                 table_name: out_rus12__monthly_demand_and_energy_at_power_sources
                 partition_expr: "EXTRACT(YEAR FROM report_date)"
+          - dbt_utils.expression_is_true:
+              arguments:
+                expression: date_trunc('month', report_date) >= date_trunc('month', peak_demand_date)
+              description: Expect peak demand date reported prior to report date.
```
Comment (aesharpe) [inline dbt/models/rus12/out_rus12__monthly_demand_and_energy_at_power_sources/schema.yml]: non-blocking: you could also add something about the year needing to be the same?

File: dbt/models/rus7/core_rus7__yearly_energy_purchased/schema.yml [hunk 2]
```diff
@@ -13,8 +13,29 @@ sources:
...
+                  config:
+                    error_if: ">99"
           - name: average_energy_cost_dollars_per_mwh
+            data_tests:
+              - dbt_expectations.expect_column_values_to_be_between:
+                  arguments:
+                    min_value: 0.0
+                  row_condition: utility_name_eia != '*Adjustments'
+                  config:
+                    error_if: ">30"
```
Comment (aesharpe) [inline dbt/models/rus7/core_rus7__yearly_energy_purchased/schema.yml:38]: Initially I didn't add a test here because I wasn't sure whether negative values were truly invalid -- i.e., maybe some special accounting causes `purchased_mwh` to be negative. But I think it's ok to include even if we are not sure because it should still be a small portion of the overall results.

File: dbt/models/rus7/core_rus7__yearly_materials_and_supplies/schema.yml [hunk 3]
```diff
@@ -9,13 +9,49 @@ sources:
               arguments:
                 table_name: core_rus7__yearly_materials_and_supplies
                 partition_expr: "EXTRACT(YEAR FROM report_date)"
+          - dbt_utils.expression_is_true:
+              arguments:
+                expression: materials_purchased + materials_salvaged - materials_sold - materials_used + materials_adjustment <= materials_ending_balance
+              description: "Check that annual changes don't exceed ending balance."
+              config:
+                error_if: ">3"
```
Comment (aesharpe) [inline dbt/models/rus7/core_rus7__yearly_materials_and_supplies/schema.yml:17]: I think that this doesn't have to be true. Ending balance is just the starting balance plus all the materials purchased minus materials sold/used, so in theory the ending balance could be lower than the sum of these components because it's just about how much inventory they have.

File: dbt/models/rus7/core_rus7__yearly_materials_and_supplies/schema.yml [hunk 4]
```diff
@@ -9,13 +9,49 @@ sources:
...
           - name: borrower_id_rus
           - name: electric_or_other_materials
           - name: materials_adjustment
           - name: materials_ending_balance
+            data_tests:
+              - dbt_expectations.expect_column_values_to_be_between:
+                  arguments:
+                    min_value: 0.0
+                  config:
+                    error_if: ">12"
```
Comment (aesharpe) [inline dbt/models/rus7/core_rus7__yearly_materials_and_supplies/schema.yml:29]: It seems unlikely but plausible that this could be negative.

File: dbt/models/rus7/out_rus7__yearly_energy_purchased/schema.yml [hunk 5]
```diff
@@ -15,8 +15,29 @@ sources:
...
+                  config:
+                    error_if: ">99"
           - name: average_energy_cost_dollars_per_mwh
+            data_tests:
+              - dbt_expectations.expect_column_values_to_be_between:
+                  arguments:
+                    min_value: 0.0
+                  row_condition: utility_name_eia != '*Adjustments'
+                  config:
+                    error_if: ">30"
```
Comment (aesharpe) [inline dbt/models/rus7/out_rus7__yearly_energy_purchased/schema.yml:40]: Adjustments seems like a field that could in theory be negative?

Comment (cmgosnell) [reply dbt/models/rus7/out_rus7__yearly_energy_purchased/schema.yml:40]: probably but the row_condition here is removing them from this test

File: dbt/models/rus7/core_rus7__yearly_materials_and_supplies/schema.yml (hunk 4 above)
Comment (cmgosnell) [reply dbt/models/rus7/core_rus7__yearly_materials_and_supplies/schema.yml:29]: unlikely like an edge case (a util liquidating materials?). so possible but still good imo to test that it is generally positive

File: dbt/models/rus7/core_rus7__yearly_materials_and_supplies/schema.yml (hunk 3 above)
Comment (cmgosnell) [reply dbt/models/rus7/core_rus7__yearly_materials_and_supplies/schema.yml:17]: hm at first i agreed austen here. my understanding about the relationships between these columns is something like this:

```
materials_ending_balance (from previous year aka starting balance) + materials_purchased + materials_salvaged - materials_sold - materials_used + materials_adjustment = materials_ending_balance
```

But after thinking about it for a second it does seem like all of the changes across the year shouldn't be more than the ending balance. I was initially thinking of the edge case of a company liquidating... but even in that context the yearly changes would presumably be larger and negative drawing the ending balance down so this would still hold. okay... after thinking about it I like your instinct here @aesharpe  but i think this is a worthwhile thing to expect and test

File: dbt/models/rus7/out_rus7__yearly_energy_purchased/schema.yml (hunk 5 above)
Comment (aesharpe) [reply dbt/models/rus7/out_rus7__yearly_energy_purchased/schema.yml:40]: oh I see, it's != my bad!

File: dbt/models/rus12/out_rus12__monthly_demand_and_energy_at_power_sources/schema.yml (hunk 1 above)
Comment (e-belfer) [reply dbt/models/rus12/out_rus12__monthly_demand_and_energy_at_power_sources/schema.yml]: Done!

File: dbt/models/rus7/core_rus7__yearly_materials_and_supplies/schema.yml (hunk 3 above)
Comment (e-belfer) [reply dbt/models/rus7/core_rus7__yearly_materials_and_supplies/schema.yml:17]: starting balance + changes = ending balance --> changes = ending balance - starting balance

If the starting balance were zero, the changes would be exactly equal to the ending balance. In any other context, if there's liquidation then we have a negative value < ending balance, and if there's a positive value we're still winding up with the changes < ending balance. A little confusing, but that's my logic for the test.

File: dbt/models/rus12/core_rus12__monthly_demand_and_energy_at_power_sources/schema.yml [hunk 6]
```diff
@@ -8,6 +8,10 @@ sources:
               arguments:
                 table_name: core_rus12__monthly_demand_and_energy_at_power_sources
                 partition_expr: "EXTRACT(YEAR FROM report_date)"
+          - dbt_utils.expression_is_true:
+              arguments:
+                expression: date_trunc('month', report_date) >= date_trunc('month', peak_demand_date) AND date_trunc('year', report_date) == date_trunc('year', peak_demand_date)
+              description: Expect peak demand date reported prior to report date.
```
Comment (aesharpe) [inline dbt/models/rus12/core_rus12__monthly_demand_and_energy_at_power_sources/schema.yml:14]: quick nit:


```suggestion
              description: Expect peak demand date reported prior to report date but within the same year.
```

File: dbt/models/rus12/out_rus12__monthly_demand_and_energy_at_power_sources/schema.yml [hunk 7]
```diff
@@ -8,6 +8,10 @@ sources:
               arguments:
                 table_name: out_rus12__monthly_demand_and_energy_at_power_sources
                 partition_expr: "EXTRACT(YEAR FROM report_date)"
+          - dbt_utils.expression_is_true:
+              arguments:
+                expression: date_trunc('month', report_date) >= date_trunc('month', peak_demand_date) AND date_trunc('year', report_date) == date_trunc('year', peak_demand_date)
+              description: Expect peak demand date reported prior to report date.
```
Comment (aesharpe) [inline dbt/models/rus12/out_rus12__monthly_demand_and_energy_at_power_sources/schema.yml]:
```suggestion
              description: Expect peak demand date reported prior to report date but within the same year.
```
